### PR TITLE
Documentation: docstrings in `asab.Application`

### PR DESCRIPTION
**Pull request by mejroslav**
For the new makedocs documentation, I would like to:
- add docstrings for all the classes and public methods
- convert the existing docstrings to [Google style format](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)

### CODE REVIEW

File: asab/application.py [hunk 1]
```diff
@@ -150,11 +195,28 @@ def create_argument_parser(
 		argument_default=None,
 		conflict_handler='error',
 		add_help=True
-	):
-		'''
-		This method can be overriden to adjust argparse configuration.
-		Refer to the Python standard library to `argparse.ArgumentParser` for details of arguments.
-		'''
+	) -> argparse.ArgumentParser:
+		"""
```
Comment (mejroslav) [inline asab/application.py:199]: image

File: asab/application.py [hunk 2]
```diff
@@ -28,19 +29,56 @@
...
 
+		```python
 		class MyApplication(asab.Application):
 			def __init__(self):
 				super().__init__(modules=[asab.web.Module, asab.zookeeper.Module])
-		'''
+		```
+		"""
+
+		self.ExitCode: typing.Union[str, int]
```
Comment (mejroslav) [inline asab/application.py:68]: image

File: asab/application.py [hunk 3]
```diff
@@ -388,17 +469,20 @@ def _do_restart(self, event_name):
 		self.stop("!RESTART!")
 
 	def restart(self):
-		'''
-		Schedules a hard restart of the whole application.
+		"""
```
Comment (mejroslav) [inline asab/application.py:472]: image

File: asab/application.py (hunk 2 above)
Comment (ateska) [reply asab/application.py:68]: `typing.Union[int, str]` ... `int` is much more important

**Review by ateska — CHANGES_REQUESTED** [review]

**Pull request merged**